runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkRunnerResult.java

```diff
@@ -73,4 +76,14 @@ public String toString() {
         ", runtime=" + runtime +
         '}';
   }
+
+  @Override
+  public State cancel() throws IOException {
+    throw new UnsupportedOperationException("FlinkRunnerResult does not support cancel.");
+  }
+
+  @Override
+  public State waitToFinish(Duration duration) {
+    throw new UnsupportedOperationException("FlinkRunnerResult does not support waitToFinish.");
```

Comment:
Currently, FlinkRunner has a blocking run is a implementation details.

I think in the future, waitToFinish() will do the following:
if (options.isBlockOnRun()) {
  return;
} else {
  // wait for the pipeline to finish.
}